src/common/exif.cc

```diff
@@ -3793,6 +3791,24 @@ static void dt_remove_xmp_key(Exiv2::XmpData &xmp, const char *key)
   }
 }
 
+static void _remove_xmp_keys(Exiv2::XmpData &xmpData, const char *key)
+{
+  try
+  {
+    const std::string needle = key;
+    for(Exiv2::XmpData::iterator i = xmpData.begin(); i != xmpData.end();)
+    {
+      if(i->key().compare(0, needle.length(), needle) == 0)
+        i = xmpData.erase(i);
+      else
+        ++i;
+    }
+  }
+  catch(Exiv2::AnyError &e)
```

Comment:
Do some exceptions happen in this function? I mean if errors are not expected here, then try-catch can be removed, since the way code is written right now exceptions will be handled outside this function by printing a error and returning a failure. Or what is the reasoning here?